### PR TITLE
Introduce a CHANGELOG

### PR DESCRIPTION
**Pull request by kylef**
As we've internally discussed, this PR introduces a changelog file. This changelog files allows us to easily keep track of the changes we make, pull requests are expected to add an entry to the changelog file in the appropriate enhancements or bug fixes section.

I've written a changelog entry for the items which have been included into master.

The entries for previous releases we're copied from GitHub releases so you don't need to review their content, since these release we're already made with these release notes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  [#286](https://github.com/apiaryio/drafter/issues/286)
+
+* Errors raised from Snowcrash, the underlying parser are now treated
+  as errors. This resolves an issue where Drafter would get into an
+  infinite loop when certain circular references are found in MSON
+  Data Structures.  
+  [#255](https://github.com/apiaryio/drafter/issues/255)
+
+* Errors raised from drafter are now correctly treated as errors,
+  previously they we're exposed as warnings.
```
Comment (pksunkara) [inline CHANGELOG.md]: `they are`

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Drafter Changelog
+
+## Master
+
+### Bug Fixes
+
+* Parameters in Refract output will no longer contain a `content` value
+  when there is no sample value.  
+  [#286](https://github.com/apiaryio/drafter/issues/286)
+
```
Comment (pksunkara) [inline CHANGELOG.md]: Improved error reporting with source maps when using cdrafter

**Comment by pksunkara** [comment]
Reviewed.

File: CHANGELOG.md (hunk 1 above)
Comment (kylef) [inline CHANGELOG.md]: Shouldn't this be 'were'? 'they are' isn't past tense.

File: CHANGELOG.md (hunk 2 above)
Comment (kylef) [inline CHANGELOG.md]: There we're no user facing memory leaks, they we're purely in the tests themselves so they don't warrant going in the CHANGELOG.

#138 and #243 we're bug in Snowcrash and not Drafter, so they should go into Snowcrash changelog.

Comment (pksunkara) [inline CHANGELOG.md]: Snowcrash neither has a changelog nor getting released anymore.

Comment (kylef) [inline CHANGELOG.md]: Okay, added these two bugs to the changelog.

**Pull request merged**